source/extensions/filters/http/jwt_authn/authenticator.h

```diff
@@ -20,13 +23,11 @@ class Authenticator {
 public:
   virtual ~Authenticator() {}
 
-  // The callback interface to notify the completion event.
-  class Callbacks {
-  public:
-    virtual ~Callbacks() {}
-    virtual void onComplete(const ::google::jwt_verify::Status& status) PURE;
-  };
-  virtual void verify(Http::HeaderMap& headers, Callbacks* callback) PURE;
+  // Verify if headers satisfyies the JWT requirements. Can be limited to single provider with
+  // extract_param.
+  virtual void
+  verify(Http::HeaderMap& headers, std::vector<JwtLocationConstPtr>&& tokens,
+         std::function<void(const ::google::jwt_verify::Status& status)>&& callback) PURE;
```

Comment:
we can typedef a AuthenticatorCallback for this std::function.

Reply:
done